### PR TITLE
Return a typed error for config validation, and make errors simple

### PR DESCRIPTION
**Pull request by smarterclayton**
Will allow clients to determine when the configuration is bad.

On v < 2, we write to os.Stderr and exit 1 instead of calling glog.Fatalf

Follow on to #5016 and #3894

### CODE REVIEW

**Comment by bgrant0607** [comment]
cc @jlowdermilk

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -38,19 +42,67 @@ import (
...
+			fatal(fmt.Sprintf("Error from server: %s", err.Error()))
+		case errors.IsUnexpectedObjectError(err):
+			fatal(fmt.Sprintf("Server returned an unexpected response: %s", err.Error()))
 		}
-		if client.IsUnexpectedStatusError(err) {
-			glog.FatalDepth(1, fmt.Sprintf("Unexpected status received from server: %s", err.Error()))
+		switch t := err.(type) {
+		case *url.Error:
+			glog.V(4).Infof("Connection error: %s %s: %v", t.Op, t.URL, t.Err)
+			switch {
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/util/helpers.go]: Any reason for single case switch over if?

**Comment by jlowdermilk** [comment]
LGTM

File: pkg/kubectl/cmd/util/helpers.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/helpers.go]: I had the original two cases and removed them.  I can reduce to if, although I expect cases to be added here for other conditions eventually.

Comment (jlowdermilk) [inline pkg/kubectl/cmd/util/helpers.go]: Fine as is, if you expect cases to be added later.

**Comment by jlowdermilk** [comment]
Kicking shippable, will merge on green.

**Pull request merged**